### ISSUE TITLE
Support both i2c interfaces - Pins and StemmaQT/Qwic

### ISSUE THREAD

**Issue by tyeth** [adafruit/Wippersnapper_Boards#70]
Hi, was trying out the QTPY ESP32-S2 and reading the components examples/code and it only supports one i2c interface currently. The spec takes an array, but no idea if the website will let me scan two interfaces. It would be great if both i2c interfaces were supported on the boards that had two. 

On a similar note, I need to ask a general question but related that should really be in the component repo: the sht3x/4x support multiple i2c addresses, so can we have both enabled by default or is it only (initially) whatever the adafruit breakouts support? I did see one component with 3 possible addresses so I assume it's just the initial implementation.

**Comment by tyeth** [comment]
I've since used a StemmaQT breakout and it has detected both the SCD40 (connected by stemma) and the SHT30 (address 44 but over dupont jumpers via QT breakout), so board and connections are fine. It's a nice smooth process using wippersnapper, great work!

**Comment by brentru** [comment]
Having access to both I2C interfaces is something we want to (and will likely) support in the future. 

> On a similar note, I need to ask a general question but related that should really be in the component repo: the sht3x/4x support multiple i2c addresses, so can we have both enabled by default or is it only (initially) whatever the adafruit breakouts support? I did see one component with 3 possible addresses so I assume it's just the initial implementation.

Having a hard time understanding this q, please correct me if I missed it. For I2C sensors that have multiple possible I2C addresses, you may add extra addresses (separated by commas) to the array [here](https://github.com/adafruit/Wippersnapper_Components/blob/main/components/i2c/SHT40/definition.json#L3).

**Comment by tyeth** [comment]
Yeah you got the right end of the stick, I was I guess just pondering why the adafruit SHT 4-series breakout didn't also include both available addresses.

Looking forward to the second bus, and with a QT breakout it's fine for now.
I did want to add another board, ESP32 based, and include both i2c's, but think I'll wait until the web interface shows support for both i2c's before going down that rabbit hole.

Cheers

**Comment by brentru** [comment]
> Yeah you got the right end of the stick, I was I guess just pondering why the adafruit SHT 4-series breakout didn't also include both available addresses.


Feel free to submit a pull request adding the second i2c address to the SHT 4 series, we accept PRs! 

Marking as long term
